complete/src/main/java/hello/FileUploadController.java

```diff
@@ -45,15 +45,15 @@ public String handleFileUpload(@RequestParam("name") String name,
 			redirectAttributes.addFlashAttribute("message", "Folder separators not allowed");
 			return "redirect:/";
 		}
-		if (name.contains("/")) {
+		if (name.contains("\\")) {
 			redirectAttributes.addFlashAttribute("message", "Relative pathnames not allowed");
 			return "redirect:/";
 		}
 
 		if (!file.isEmpty()) {
 			try {
 				BufferedOutputStream stream = new BufferedOutputStream(
-						new FileOutputStream(new File(Application.ROOT + "/" + name)));
+						new FileOutputStream(new File(Application.ROOT + File.separator + name)));
                 FileCopyUtils.copy(file.getInputStream(), stream);
 				stream.close();
 				redirectAttributes.addFlashAttribute("message",
```

Comment:
Overall, this guide could do better if we used more of the Java NIO APIs (Files and Paths).